proxy/http/HttpSM.cc

```diff
@@ -665,6 +665,20 @@ HttpSM::state_read_client_request_header(int event, void *data)
     state = PARSE_RESULT_ERROR;
   }
 
+#if TS_HAS_TLS_EARLY_DATA
+  SSLNetVConnection *ssl_vc = dynamic_cast<SSLNetVConnection *>(netvc);
+  if (ssl_vc != nullptr && ssl_vc->read_from_early_data) {
+    ssl_vc->read_from_early_data = false;
+    // We check whether the methods are safe according to section 9.1.1
+    // from here https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html
```

Comment:
Picky comment: RFC2616 is obsoleted by RFC7231 :)  https://tools.ietf.org/html/rfc7231#section-4.2.1